llvm/test/tools/llvm-ml/rip_relative_addressing.asm

```diff
@@ -53,4 +53,10 @@ mov eax, [t8]
 ; CHECK-LABEL: t8:
 ; CHECK: mov eax, dword ptr [t8]
 
-END
\ No newline at end of file
+t9:
+mov eax, dword ptr [bar]
```

Comment:
This seems problematic if we paser `mov eax, [bar]` and `mov eax, dword ptr [bar]` differently.

Reply:
Unfortunately, this appears accurate to how MASM actually works! Having run the following code through ml64.exe, I do reproduce the behavior shown in this change.

In particular, with `bar` being a label in the `.data` section:
```
t1:
mov eax, [bar]
```
compiles to an OBJ file that disassembles as:
```
0000000000000000 <.text$mn>:
       0: 8b 05 00 00 00 00            	movl	(%rip), %eax            # 0x6 <.text$mn+0x6>
		0000000000000002:  IMAGE_REL_AMD64_REL32	bar
```
while
```
t2:
mov eax, dword ptr [bar]
```
compiles to an OBJ file that disassembles as:
```
0000000000000000 <.text$mn>:
       0: b8 00 00 00 00               	movl	$0x0, %eax
		0000000000000001:  IMAGE_REL_AMD64_ADDR32	bar
```

It seems MASM resolves `[bar]` as a relative offset in one case and a full address in the other. I can't entirely explain this, but it may be relevant that in MASM, `<type> PTR` is actually a cast operator. (https://learn.microsoft.com/en-us/cpp/assembler/masm/operator-ptr?view=msvc-170)

Reply:
Oh - and running the same code through LLVM-ML without this change, we get:
```
0000000000000000 <t1>:
       0: 8b 04 25 00 00 00 00         	movl	0x0, %eax
		0000000000000003:  IMAGE_REL_AMD64_ADDR32	bar
```
in both cases, which is clearly not a match for ml64.